### PR TITLE
Select multiple messages and bulk deletion

### PR DESCRIPTION
**Pull request by msgmaxim**
- select multiple messages by left-clicking on them
- when at least one message is selected it enters a *multi-select* mode with an additional menu to delete selected messages or cancel the selection.

### CODE REVIEW

File: js/models/conversations.js [hunk 1]
```diff
@@ -2420,14 +2461,23 @@
...
       }
-      const success = await channelAPI.deleteMessage(message.getServerId());
+
+      let success;
+      if (messages.length > 1) {
+        success = await channelAPI.deleteMessages(
+          messages.map(m => m.getServerId())
+        );
+      } else {
+        success = await channelAPI.deleteMessages([messages[0].getServerId()]);
```
Comment (sachaaaaa) [inline js/models/conversations.js]: looks to me like the if/else is not necessary, unless I'm mistaken?

File: js/models/conversations.js [hunk 2]
```diff
@@ -217,6 +219,43 @@
...
+      }
+
+      this.trigger('message-selection-changed');
+    },
+
+    resetMessageSelection() {
+      this.selectedMessages.clear();
+      this.messageCollection.forEach(m => {
+        // eslint-disable-next-line no-param-reassign
+        m.selected = false;
```
Comment (sachaaaaa) [inline js/models/conversations.js]: why do we need to set `selected` to false here but not in `removeMessageSelection` ?

Comment (sachaaaaa) [reply js/models/conversations.js]: oh ok, just saw that `selected` was already toggled on click at the message level

**Comment by BeaudanBrown** [comment]
Not the most pressing issue but I had a suspicion and just confirmed that you can enter *multi-select* mode by selecting some messages then use the conversation settings cog to `Delete messages`, at which point you will be stuck in multi-select mode until you restart your client because there are ids in the selected list which can't be removed

**Comment by BeaudanBrown** [comment]
Well whoops actually you can still press `delete` or `cancel`

File: js/models/conversations.js (hunk 1 above)
Comment (msgmaxim) [reply js/models/conversations.js]: Updated this part

**Pull request merged**